### DISCUSSION TITLE
function 下如何获取其他lua中的字符串变量？

### DISCUSSION THREAD

**Discussion by DanielKui** [xmake-io/xmake#3891] [General]
![image](https://github.com/xmake-io/xmake/assets/578118/5d290f3e-6613-4c8a-93d2-b1a88ed69a1e)

before_link(function (target) 里无法获取tarDir的定义，实际定义为"./bin"定义在 includes ("HXQuoteDemoFiles.lua")中

**Comment by waruqi** [comment]
先看下文档 描述域和脚本域 说明。。https://xmake.io/#/zh-cn/guide/syntax_description

两个域是隔离的，不要混用。。脚本域 用 target:targetdir() 获取

**Reply by DanielKui** [reply]
非常感谢

如果我没有set_targetdir(tarDir)这一步的话， 如何在描述域 里 获取includes ("HXQuoteDemoFiles.lua") 文件里定义的变量呢？

**Reply by waruqi** [reply]
描述域之前变量获取，直接走 全局 lua 变量。。。描述域跟脚本域之间，走  set_values("xxx"), target:values("xxx") 传递
